pre-docs/writing_plans.md

```diff
@@ -494,18 +494,36 @@ plan run_with_facts(TargetSpec $nodes) {
 }
 ```
 
-**Related information**  
+### Collect general data from PuppetDB
+
+You can use the `puppetdb_query` function in plans to make direct queries to PuppetDB. For example you can discover nodes from PuppetDB and then run tasks on them. You'll have to configure the [puppetdb client](bolt_connect_puppetdb.md)before running it. You can learn how to [structure pql queries here](https://puppet.com/docs/puppetdb/latest/api/query/tutorial-pql.html), and find [pql reference and examples here](https://puppet.com/docs/puppetdb/latest/api/query/v4/pql.html)
+
+```
+plan pdb_discover {
+  $result = puppetdb_query("inventory[certname] { app_role == 'web_server' }")
+  # extract the certnames into an array
+  $names = $result.map |$r| { $r["certname"] }
+  # wrap in url. You can skip this if the default transport is pcp
+  $nodes = $names.map |$n| { "pcp://${n}" }
+  run_task('my_task', $nodes)
+}
+```
 
+**Related information**  
 
 [Connecting Bolt to PuppetDB](bolt_connect_puppetdb.md)
 
 ## Plan logging
 
-Set up log files to record certain events that occur when you run plans.
+Plan information can be captured in log files or printed to a terminal session. This section details the methods for collecting information about plan runs.
+
+### Outputting message to the terminal
+
+The plan function [`out::message`](plan_functions.md#outmessage) can be used to print message strings to `STDOUT`. These message will always be printed regardless of the log level and will not be logged to the log file.
```

Comment:
it looks like we're not printing it with format=json yet. 
https://github.com/puppetlabs/bolt/blob/master/lib/bolt/outputter/json.rb#L109
we should really fix that.

Reply:
Created https://tickets.puppetlabs.com/browse/BOLT-1455 - feel free to add more detail!